documentation/modules/oauth/con-mapping-keycloak-authz-services-to-kafka-model.adoc

```diff
@@ -0,0 +1,92 @@
+[id="con-mapping-keycloak-authz-services-to-kafka-model_{context}"]
+= Mapping Keycloak Authorization Services to the Kafka authorization model
+
+[role="_abstract"]
+
+To grant Kafka permissions to user accounts or service accounts, you first create an OAuth _client_ specification for the Kafka broker in Keycloak. You then specify Keycloak Authorization Services rules on the client.
+Typically, the client id of the broker OAuth client is `kafka`.
+
+The `kafka` client definition must have the *Authorization Enabled* option enabled in the Keycloak Admin Console.
+
+All permissions exist within the scope of the `kafka` client. If you have different Kafka clusters configured with different OAuth client IDs, they each need a separate set of permissions even though they're part of the same Keycloak realm.
+
+When the Kafka client uses OAUTHBEARER authentication, the Keycloak authorizer (`KeycloakRBACAuthorizer`) uses the access token of the current session to retrieve a list of grants from the Keycloak server.
+To retrieve the grants, the authorizer evaluates the Keycloak Authorization Services policies and permissions.
+
+.Authorization scopes for Kafka permissions
+
+An initial Keycloak configuration usually involves uploading authorization scopes to create a list of all possible actions that can be performed on each Kafka resource type.
+This step is performed once only, before defining any permissions.
+You can add authorization scopes manually instead of uploading them.
+
+Authorization scopes must contain all the possible Kafka permissions regardless of the resource type:
+
+* `Create`
+* `Write`
+* `Read`
+* `Delete`
+* `Describe`
+* `Alter`
+* `DescribeConfig`
+* `AlterConfig`
+* `ClusterAction`
+* `IdempotentWrite`
+
+.Resource patterns for permissions checks
+
+Resource patterns are used for pattern matching against the targeted resources when performing permission checks.
+The general pattern format is `__RESOURCE-TYPE:PATTERN-NAME__`.
+
+The resource types mirror the Kafka authorization model.
+The pattern allows for two matching options:
+
+* Exact matching (when the pattern does not end with `\*`)
```

Comment:
@mstruk - Is `\*` a typo? Should it say does not end with * ?

Reply:
Looks to me like an escape char for Markdown. The text should display only asterisk.